modules/base/src/time/calendar_duration.fz

```diff
@@ -90,7 +90,48 @@ is
   # create a string representation of this duration.
   #
   public redef as_string String =>
-    "NYI"
+    lm : mutate.
+    lm ! ()->
+      is_positive := lm.env.new bool false
```

Comment:
I would prefer not using a mut here but `is_positive := years > 0 || months >0 || ...`